### PR TITLE
fix #175031: selectable palette in Note Group UI.

### PR DESCRIPTION
**Pull request by ericfont**
Clicking notes toggles between palette's currently selected beam mode and auto.
Initial default selected beam mode is BEGIN, which will keep default behavior similar behavior prior to this commit.

### CODE REVIEW

File: mscore/noteGroups.cpp [hunk 1]
```diff
@@ -86,6 +86,8 @@ NoteGroups::NoteGroups(QWidget* parent)
       iconPalette->setDrawGrid(true);
       populateIconPalette(iconPalette, bpa);
       iconPalette->setReadOnly(true);
+      iconPalette->setSelectable(true);
+      iconPalette->setSelected(0);
```
Comment (ericfont) [inline mscore/noteGroups.cpp]: This sets initial selected state of palette to Beam::Mode::BEGIN.

File: mscore/noteGroups.cpp [hunk 2]
```diff
@@ -139,6 +141,8 @@ Groups NoteGroups::groups()
 void NoteGroups::resetClicked()
       {
       setSig(_sig, _groups);
+      iconPalette->setSelected(0);
+      iconPalette->update();
```
Comment (ericfont) [inline mscore/noteGroups.cpp]: have to force update here to trigger redraw of palette reflect resetting selecting beam mode back to the default.

File: mscore/noteGroups.cpp [hunk 3]
```diff
@@ -148,10 +152,31 @@ void NoteGroups::resetClicked()
...
+            default:
+                  selectedBeamMode = Beam::Mode::BEGIN;
+                  break;
+            }
+
+      if (chord->beamMode() != selectedBeamMode)
+            chord->setBeamMode(selectedBeamMode);
+      else
             chord->setBeamMode(Beam::Mode::AUTO);
+
```
Comment (ericfont) [inline mscore/noteGroups.cpp]: note that behavior of above if/else is slightly different than was previously, but I much prefer my new behavior.  With the old version, clicking on a note that wasn't either AUTO or BEGIN would do nothing, so I'm afraid user would be confused and think that the UI was simply not working.  With my new behavior, clicking on a note will always do something to the beam mode.

**Comment by lasconic** [comment]
See https://musescore.org/fr/node/175031#comment-660186
It also needs to be rebased if we wanted to merge it.

**Comment by ericfont** [comment]
Rebased and made a few changes: 
- Increased the clickable region such that click event will change beam mode of nearest note (so can click on whitespace).
- Clicking notes toggles between palette's currently selected beam mode and auto.
- Initial default keeps prior behavior whereby togging works the same (I'm using the same if statement)
- If an palette cell is selected, then click on notes will change to the selected beam property.  (So no longer will toggle if a palette cell is selected).
- Reset button will now remove the palette's selection.
- Removed the unnecessary labels "1/8", "1/16", "1/32" from the UI.
- Moved the hspacer on the UI to be to the right of the palette.

**Comment by dmitrio95** [comment]
Needs actualization for current master.

**Pull request closed without merging**